### PR TITLE
Split out docs from package

### PR DESCRIPTION
**Pull request by davidkretch**
## Main changes
* Remove full docs from the CRAN packages, and add a link to the web docs.
* Write full docs only to the `paws` folder, from which we can generate the web docs. The workflow for generating the web docs remains the same.

## Other changes
* Update packages.yml to use the package name as it appears in the .json filename, e.g. compute-optimizer rather than computeoptimizer. This should make it easier to update.
* Remove code for the no-longer-needed sub-category packages.
* Do not stop on code generation cache mismatches. Code generation often fails due to spurious mismatches.
* Combine the code generation and CRAN package generation into a single function and Make recipe. This allows us to generate only the code we need to build the CRAN packages.

### CODE REVIEW

**Comment by davidkretch** [comment]
Hello @DyfanJones, I think this is ready.  Let me know what you think.

**Comment by DyfanJones** [comment]
> Hello @DyfanJones, I think this is ready. Let me know what you think. I recognize it's a large change -- sorry.

No worries, I will review it tomorrow :D

**Comment by DyfanJones** [comment]
Do this change any of the current [DEVELOPER_GUIDE.md](https://github.com/paws-r/paws/blob/main/DEVELOPER_GUIDE.md) documentation?

**Comment by DyfanJones** [comment]
It looks all good to me, just got one or two questions around the Developer guides (if they need an update) and the TODO comment in the unit test :D

File: make.paws/R/process_api.R [hunk 1]
```diff
@@ -1,60 +1,104 @@
...
+
+#-------------------------------------------------------------------------------
+
+# Write a dictionary of key value pairs to disk, where the key is the file name
+# and the value is the contents of the file. If the contents are themselves a
+# dictionary, recursively write the dictionary.
+write_dict <- function(dict, dir) {
+  for (file in names(dict)) {
+    contents <- dict[[file]]
+    if (class(contents) == "character") {
```
[the remark this answers is not in the document]
Comment (davidkretch) [reply make.paws/R/process_api.R]: Thanks. Addressed.

File: make.paws/tests/testthat/test_make_sdk.R [hunk 2]
```diff
@@ -83,7 +83,9 @@ path_out <- file.path(path, "out")
 #-------------------------------------------------------------------------------
 
 test_that("make_sdk", {
-  make_sdk(path_in, path_out)
+  # TODO(davidkretch): Fix test.
```
[the remark this answers is not in the document]
Comment (davidkretch) [reply make.paws/tests/testthat/test_make_sdk.R]: Got it. Addressed.

**Comment by davidkretch** [comment]
> Do this change any of the current [DEVELOPER_GUIDE.md](https://github.com/paws-r/paws/blob/main/DEVELOPER_GUIDE.md) documentation?

It looks like the developer guide is ok as is.

**Comment by DyfanJones** [comment]
Looks all good to me, feel free to merge when ready :)

**Comment by davidkretch** [comment]
> Looks all good to me, feel free to merge when ready :)

Sounds good, thanks!

**Pull request merged**